_exercises/advanced/arrays/2-reduce.js

```diff
@@ -17,27 +17,52 @@ const customers = require('../../../data/customers');
 
     Use the reduce method
  */
+console.log  ( "  1. Display the list of customers as an object that looks like this { name: balance }") ;
+
+const customerObject = customers.reduce( (obj, customer) => {
+    obj[customer.name.first + " " + customer.name.last] = customer.balance ;
+    return obj ;
+},{}) ;
+
+console.log ( customerObject) ;
 
 /*
 
     2. Display the full names of all the males that are over 30
 
  */
 
+console.log  ( " 2. Display the full names of all the males that are over 30") ;
+const isAbove30 = (item) => item.age > 30 ;
+const isMale = (item) => item.isMale ;
+const asFullName = (person) => person.name.last + " " + person.name.first ;
+console.log ( customers.filter(isAbove30 && isMale).map(asFullName))  ;
```

Comment:
Does this work ??

When looking at the code I would think this is not working correctly.